HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -635,23 +636,47 @@ def check_for_errors
         end
       end
 
+      if not @dst_begin_month.nil?
```

Comment:
Rather than copying/pasting new sections of code only to make a few small changes, update the existing start/end code to also check DST. You can either loop through both or create a method to accomplish this.